lib/GADS.pm

```diff
@@ -2753,6 +2753,235 @@ prefix '/:layout_name' => sub {
         template 'data' => $params;
     };
 
+    prefix '/report' => sub {
+
+        #view all reports for this instance
+        get '' => require_login sub {
+            my $user   = logged_in_user;
+            my $layout = var('layout') or pass;
+
+            my %params = (
+                user   => $user,
+                layout => $layout,
+                schema => schema,
+            );
+
+            my $records = GADS::Records->new(%params);
+
+            my $alert = GADS::Alert->new(
+                user   => $user,
+                layout => $layout,
+                schema => schema,
+            );
+
+            my $base_url = request->base;
+
+            my $params;
+
+            $params->{alerts}      = $alert->all;
+            $params->{header_type} = 'table_tabs';
+
+            $params->{layout_obj} = $layout;
+            $params->{layout}     = $layout;
+
+            $params->{header_back_url} = "${base_url}table";
+            $params->{breadcrumbs}     = [
+                Crumb( $base_url . "table/", "Tables" ),
+                Crumb( "",                   "Table: " . $layout->name )
+            ];
+
+            my $layout_id = $layout->{instance_id};
+
+            my $reports = $layout->reports;
+
+            $params->{reports} = $reports;
+
+            template 'reports/view' => $params;
+        };
+
+        #add a report
+        any [ 'get', 'post' ] => '/add' => require_login sub {
+            if ( body_parameters && body_parameters->{submit} ) {
+                my $layout             = var('layout') or pass;
+                my $report_description = params->{'report_description'};
+                my $report_name        = params->{'report_name'};
+                my $checkbox_fields    = params->{'checkboxes'};
```

Comment:
Can you change `params` to `body_parameters->get()` (or `get_all`) please? Using body_parameters ensures that URL query parameters are not used, and it also ensures a consistent type of variable (e.g. if you try submitting a report with only one field ticked you will get an error, as the array becomes a scalar without the use of `get_all`